### ISSUE TITLE
[ENH] Extract mask coverage per parcel as a pandas DataFrame

### ISSUE THREAD

**Issue by claraElk** [nilearn/nilearn#6046]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe your proposed enhancement in detail.

Currently, the relative size (in %) of each region is only available in the HTML report. It would be useful to expose this information as a pandas DataFrame, making it easier to access for downstream use. 

Also, it could be useful to also extract the mask coverage per parcel (i.e. the percentage of available voxels within each parcel).

### Benefits to the change

- Using mask coverage per region could serve as a sanity check indicator of data quality

### Pseudocode for the new behavior, if applicable

```python
# insert your code below

```

**Comment by claraElk** [comment]
Here is a snippet of the current code:
 
https://github.com/nilearn/nilearn/blob/b029fb89b8d17596263a2018c3d70ff60682b8b0/nilearn/maskers/nifti_labels_masker.py#L392-L407

**Comment by htwangtw** [comment]
Related issue: #5476

**Comment by pbergeret12** [comment]
One possibility to quantify the actual coverage of the atlas by the subject's BOLD signal would be to generate a boolean mask from the 4D functional volume, and then calculate the percentage of the atlas weights that actually fall within this mask.

Here is a code example using the MSDL atlas and data from this tutorial: lhttps://nilearn.github.io/stable/auto_examples/03_connectivity/plot_inverse_covariance_connectome.html: 

```
import numpy as np
import pandas as pd
from nilearn.masking import compute_epi_mask
from IPython.display import display

# 1. Compute the data mask 
data_mask_img = compute_epi_mask(data.func[0])
mask_data = data_mask_img.get_fdata().astype(bool)

# 2. Retrieve atlas maps (already resampled by the masker)
maps_data = masker.maps_img_.get_fdata()

# 3. Calculate coverage ratio (weights inside mask / total weights)
total_weights = np.sum(np.abs(maps_data), axis=(0, 1, 2))
masked_weights = np.sum(np.abs(maps_data[mask_data, :]), axis=0)
coverage_ratio = np.divide(masked_weights, total_weights, where=total_weights != 0)

# 4. Display final table
df_coverage = pd.DataFrame({
    'Region': labels, 
    'Coverage %': coverage_ratio * 100
})
display(df_coverage)
```

**Comment by Remi-Gau** [comment]
note that after fitting, label maskers already contain a "lut_" that gives you access to a look-up table (generated at fit time or adapted from some of the arguments passed at initialization) as pandas dataframe that already contains bare mimimum of information for each region:

```python
from nilearn.datasets import fetch_atlas_harvard_oxford, fetch_development_fmri
from nilearn.maskers import NiftiLabelsMasker

dataset = fetch_atlas_harvard_oxford("cort-maxprob-thr25-2mm")
atlas_filename = dataset.maps
labels = dataset.labels
look_up_table = dataset.lut

data = fetch_development_fmri(n_subjects=1, reduce_confounds=True)
fmri_filenames = data.func[0]

masker = NiftiLabelsMasker(
    labels_img=atlas_filename,
    lut=look_up_table,
    memory="nilearn_cache",
    standardize=None
)

masker.fit_transform(fmri_filenames)

print(masker.lut_)
```

so I think we could just update the content of that existing dataframe rather than add another one
